File: Zadania z PDF'ów/Szyfr Cezara/c++/Szyfrowanie tekstu wczytanego z pliku tekstowego.cpp
#include <iostream>
#include <string>
#include <fstream>

using namespace std;

string Cezar(string s, int klucz)
{
    string szyfrogram = "";
    for (int i = 0; i < s.length(); i++) {
        if (s[i] >= 'A' && s[i] <= 'Z') {
            szyfrogram += (s[i] - 'A' + klucz) % 26 + 'A';
        } else if (s[i] >= 'a' && s[i] <= 'z') {
            szyfrogram += (s[i] - 'a' + klucz) % 26 + 'a';
        }
    }
    return szyfrogram;
}
int main()
{
    ifstream wejscie("C:/dane.txt");
    ofstream wyjscie("C:/wynik.txt");
    string s;
    int klucz;

    cout << "Podaj klucz: ";
    cin >> klucz;

    while (getline(wejscie, s))
    {
        wyjscie << Cezar(s, klucz) << endl;
    }

    wejscie.close();
    wyjscie.close();

    cout << "Plik zostal utworzony.";
    return 0;
}

File: Zadania z PDF'ów/Szyfr Cezara/Python/Szyfrowanie tekstu wczytanego z pliku tekstowego.py
import string

try:
    with open("dane.txt", "r") as f:
        wejscie = f.read()
except FileNotFoundError:
    print("Nie znaleziono pliku wejsciowego!")

wyjscie = ""
klucz = 0

klucz = int(input("Podaj klucz: "))

for i in range(0, len(wejscie)):
    if wejscie[i] == " ":
        wyjscie = wyjscie + " "
    elif wejscie[i] >= 'A' and wejscie[i] <= 'Z':
        wyjscie += chr((ord(wejscie[i]) - ord('A') + klucz) % 26 + ord('A'))
    elif wejscie[i] >= 'a' and wejscie[i] <= 'z':
        wyjscie += chr((ord(wejscie[i]) - ord('a') + klucz) % 26 + ord('a'))

try:
    with open("wyjscie.txt", "w") as f:
        f.write(wyjscie)
        f.close()
except FileNotFoundError:
    print("Nie znaleziono pliku wyjsciowego!")



